packages/langium-cli/test/generator/ast-generator.test.ts

```diff
@@ -0,0 +1,836 @@
+        `;
+        const expectedASTFile = normalizeEOL(expandToString`
+        /******************************************************************************
+        * This file was generated by langium-cli 1.1.0.
+        * DO NOT EDIT MANUALLY!
+        ******************************************************************************/
+       
+       /* eslint-disable */
+       import { AstNode, AbstractAstReflection, ReferenceInfo, TypeMetaData } from 'langium';
+       
+       export type A = 'a';
+       
+       export function isA(item: string): item is A {
+           return item === 'a';
+       }
+       
+       export type testAstType = {
+       }
+
+       export class testAstReflection extends AbstractAstReflection {
+       
+           getAllTypes(): string[] {
+               return [];
+           }
+       
+           protected override computeIsSubtype(subtype: string, supertype: string): boolean {
+               switch (subtype) {
+                   default: {
+                       return false;
+                   }
+               }
+           }
+       
+           getReferenceType(refInfo: ReferenceInfo): string {
+               const referenceId = \`\${refInfo.container.$type}:\${refInfo.property}\`;
+               switch (referenceId) {
+                   default: {
+                       throw new Error(\`\${referenceId} is not a valid reference id.\`);
+                   }
+               }
+           }
+       
+           getTypeMetaData(type: string): TypeMetaData {
+               switch (type) {
+                   default: {
+                       return {
+                           name: type,
+                           mandatory: []
+                       };
+                   }
+               }
+           }
+       }
+       
+       export const reflection = new testAstReflection();
+       
+       `).trimStart();
+        const result = (await parseHelper<Grammar>(services.grammar)(testGrammar)).parseResult;
+        const config: LangiumConfig = {
+            [RelativePath]: './',
+            projectName: 'test',
+            languages: []
+        };
+        const typesFileContent = generateAst(services.grammar, [result.value], config);
+        expect(typesFileContent).toEqual(expectedASTFile);
+    });
+
+    test('should generate checker functions for datatype rules comprised of a multiple strings', async () => {
+        const testGrammar = `
+            grammar TestGrammar
+            
+            A returns string:
+                'a' | 'b' | 'c';
+
+            hidden terminal WS: /\\s+/;
+            terminal ID: /[_a-zA-Z][\\w_]*/;
+            hidden terminal ML_COMMENT: /\\/\\*[\\s\\S]*?\\*\\//;
+            hidden terminal SL_COMMENT: /\\/\\/[^\\n\\r]*/;
+        `;
+        const expectedASTFile = normalizeEOL(expandToString`
+        /******************************************************************************
```

Comment:
Suggestion: These type of tests should be performed as isolated as possible, so that unrelated changes don't affect the test. You should cut the unrelated parts off using something like this:

```ts
const relevantPart = generatedCode.substring(
  generatedCode.indexOf('export') // from first exported type...
  generatedCode.indexOf('export class') // ... until exported reflection class
);
```

That way, we only test the generated types declarations